vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java

```diff
@@ -97,7 +98,10 @@ public void exportResults() throws IOException
             }
         }
 
-        addTestCasesToTestExecution(testCaseIds);
+        consumeOnNotNull(xrayExporterOptions.getTestExecutionKey(),
+            key -> xrayFacade.updateTestExecution(key, testCaseIds));
```

Comment:
I believe it would be logically correct to add tests to test sets first, and then to add them to tests execution